magnet/src/main/java/magnet/internal/InstanceBucket.java

```diff
@@ -218,6 +273,16 @@ void putSingle(@NotNull InstanceBucket.SingleObjectInstance<T> single) {
             //noinspection SuspiciousMethodCalls
             return instances.containsKey(factory.getClass());
         }
+
+        public boolean accept(ScopeVisitor visitor) {
+            Collection<SingleObjectInstance<T>> singleObjectInstances = instances.values();
+            boolean takeNext = true;
+            for (SingleObjectInstance<T> instance : singleObjectInstances) {
+                takeNext = instance.accept(visitor);
```

Comment:
What could be the reason for a single object instance to not accept the visitor and - as a result - stop visiting all the other instances that come after it? I don't get this use case.

Reply:
That's a search case. Find and stop further processing.

Reply:
Ah, ok, read this wrong then, sorry!